OptimizeIndexJob fails with NPE

```
2021-02-22 18:08:01,651 ERROR: org.graylog2.system.jobs.SystemJobManager - Unhandled error while running SystemJob <851359d0-7530-11eb-8601-623a1b0b38aa> [org.graylog2.indexer.indices.jobs.OptimizeIndexJob]
org.graylog.shaded.elasticsearch7.org.elasticsearch.ElasticsearchException: An error occurred: 
	at org.graylog.storage.elasticsearch7.ElasticsearchClient.exceptionFrom(ElasticsearchClient.java:136) ~[classes/:?]
	at org.graylog.storage.elasticsearch7.ElasticsearchClient.execute(ElasticsearchClient.java:99) ~[classes/:?]
	at org.graylog.storage.elasticsearch7.ElasticsearchClient.execute(ElasticsearchClient.java:92) ~[classes/:?]
	at org.graylog.storage.elasticsearch7.IndicesAdapterES7.optimizeIndex(IndicesAdapterES7.java:400) ~[classes/:?]
	at org.graylog2.indexer.indices.Indices.optimizeIndex(Indices.java:326) ~[classes/:?]
	at org.graylog2.indexer.indices.jobs.OptimizeIndexJob.execute(OptimizeIndexJob.java:71) ~[classes/:?]
	at org.graylog2.system.jobs.SystemJobManager$1.run(SystemJobManager.java:89) [classes/:?]
	at com.codahale.metrics.InstrumentedScheduledExecutorService$InstrumentedRunnable.run(InstrumentedScheduledExecutorService.java:241) [metrics-core-4.1.9.jar:4.1.9]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run$$$capture(FutureTask.java:264) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java) [?:?]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
Caused by: java.lang.NullPointerException
	at org.graylog.shaded.elasticsearch7.org.apache.http.client.config.RequestConfig.copy(RequestConfig.java:383) ~[elasticsearch7-7.9.1-0.jar:?]
	at org.graylog.storage.elasticsearch7.ElasticsearchClient.withTimeout(ElasticsearchClient.java:153) ~[classes/:?]
	at org.graylog.storage.elasticsearch7.IndicesAdapterES7.lambda$optimizeIndex$22(IndicesAdapterES7.java:400) ~[classes/:?]
	at org.graylog.storage.elasticsearch7.ElasticsearchClient.execute(ElasticsearchClient.java:97) ~[classes/:?]
	... 13 more
```
	

## Your Environment
<!--- Include as many relevant details about the environment you experienced the bug in -->

* Graylog Version: 7b2df771c9  (February 22)
* Java Version:
* Elasticsearch Version: 7.10.4
* MongoDB Version:
* Operating System:
* Browser version:

--- a/graylog-storage-elasticsearch7/src/main/java/org/graylog/storage/elasticsearch7/ElasticsearchClient.java
+++ b/graylog-storage-elasticsearch7/src/main/java/org/graylog/storage/elasticsearch7/ElasticsearchClient.java
@@ -150,13 +150,16 @@ public class ElasticsearchClient {
     }
 
     public static RequestOptions withTimeout(RequestOptions requestOptions, Duration timeout) {
-        final RequestConfig.Builder requestConfigBuilder = requestOptions.getRequestConfig() == null
+        final RequestConfig.Builder requestConfigBuilder = (requestOptions == null || requestOptions.getRequestConfig() == null)
                 ? RequestConfig.custom()
                 : RequestConfig.copy(requestOptions.getRequestConfig());
         final RequestConfig requestConfigWithTimeout = requestConfigBuilder
                 .setSocketTimeout(Math.toIntExact(timeout.toMilliseconds()))
                 .build();
-        return requestOptions.toBuilder()
+        final RequestOptions.Builder requestOptionsBuilder = requestOptions == null
+                ? RequestOptions.DEFAULT.toBuilder()
+                : requestOptions.toBuilder();
+        return requestOptionsBuilder
                 .setRequestConfig(requestConfigWithTimeout)
                 .build();
 

--- a/graylog-storage-elasticsearch7/src/main/java/org/graylog/storage/elasticsearch7/ElasticsearchClient.java
+++ b/graylog-storage-elasticsearch7/src/main/java/org/graylog/storage/elasticsearch7/ElasticsearchClient.java
@@ -150,7 +150,10 @@ public class ElasticsearchClient {
     }
 
     public static RequestOptions withTimeout(RequestOptions requestOptions, Duration timeout) {
-        final RequestConfig requestConfigWithTimeout = RequestConfig.copy(requestOptions.getRequestConfig())
+        final RequestConfig.Builder requestConfigBuilder = requestOptions.getRequestConfig() == null
+                ? RequestConfig.custom()
+                : RequestConfig.copy(requestOptions.getRequestConfig());
+        final RequestConfig requestConfigWithTimeout = requestConfigBuilder
                 .setSocketTimeout(Math.toIntExact(timeout.toMilliseconds()))
                 .build();
         return requestOptions.toBuilder()

--- a/graylog2-server/src/test/java/org/graylog2/indexer/indices/IndicesIT.java
+++ b/graylog2-server/src/test/java/org/graylog2/indexer/indices/IndicesIT.java
@@ -16,6 +16,7 @@
  */
 package org.graylog2.indexer.indices;
 
+import com.github.joschi.jadconfig.util.Duration;
 import com.google.common.collect.ImmutableMap;
 import com.google.common.collect.ImmutableSet;
 import com.google.common.eventbus.EventBus;
@@ -447,5 +448,12 @@ public abstract class IndicesIT extends ElasticsearchBaseTest {
         importFixture("org/graylog2/indexer/indices/IndicesIT.json");
 
         assertThat(indices.numberOfMessages("graylog_0")).isEqualTo(10);
+    }
+
+    @Test
+    public void optimizeIndexJobDoesNotThrowException() {
+        importFixture("org/graylog2/indexer/indices/IndicesIT.json");
+
+        indices.optimizeIndex("graylog_0", 1, Duration.minutes(1));
     }
 }

